packages/cashier/src/hooks/useWS.ts

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+import { useWS as useWSShared } from '@deriv/shared';
+import { TSocketCallTypes } from 'Types';
+
+type TSocketCalls = keyof TSocketCallTypes;
+
+type TResponse<T extends TSocketCalls> = TSocketCallTypes[T]['response'][T];
+
+type TRequestPropsExceptions = 'verify_email';
+
+type TRequestProps<T extends TSocketCalls> = Omit<
+    TSocketCallTypes[T]['request'],
+    (T extends TRequestPropsExceptions ? never : T) | 'passthrough' | 'req_id'
```

Comment:
```suggestion
    (TSocketCallTypes[T]['request'][T] extends 1 ? never : T) | 'passthrough' | 'req_id'
```
Should’ve been something like this to make it dynamic and remove unnecessary `TRequestPropsExceptions`, But I couldn’t get it working.